### PR TITLE
Feature numpy queue implementation

### PR DESCRIPTION
**Pull request by BrandonKroes**
- Yes, I know. Bedtime.

- The numpy and the queue implementation are done.

### CODE REVIEW

File: module/Algorithms.py [hunk 1]
```diff
@@ -1,12 +1,18 @@
...
 def calculate_heuristic(neighbor, end):
     """
     Calculating the heuristic based on coordinates of end and next cell
     :param neighbor: cell towards the end
     :param end:  cell where we want to end up
     :return: int value for the distance
     """
-    distance = abs(neighbor.x - end.x) + abs(neighbor.y - end.y)
-    return distance
+    if None == neighbor or None == end:
```
Comment (nicovanbentum) [inline module/Algorithms.py]: should be "var == None" or "not var"

File: module/Algorithms.py [hunk 2]
```diff
@@ -211,3 +215,109 @@ def run(self):
...
+        self.end = end
+        self.start = start
+        self.path = []
+        self.current_cell = None
+
+        """In here you will find one of the biggest changes made in the implementation
+        NumPy is the master when it comes to data management, so we won't even bother creating unique instances
+        for each and every cell, we will leave that to numpy. What we do need to do is keep track of the data
+        We would ordinarily store in a cell, in this case we use 
+        """
```
Comment (nicovanbentum) [inline module/Algorithms.py:251]: unfinished doc?

File: module/Grid.py [hunk 3]
```diff
@@ -113,6 +112,34 @@ def __str__(self):
         return text
 
 
+class GridInNumpy(Grid):
+    """The entire numpy grid should be"""
```
Comment (nicovanbentum) [inline module/Grid.py]: unfinished doc?

**Comment by BrandonKroes** [comment]
Doc issues and yoda operators fixed

**Review by nicovanbentum — APPROVED** [review]

**Pull request merged**